codemate/block.py

```diff
@@ -0,0 +1,378 @@
+     A generator of a Python block syntax.
+
+     The block syntax structure:
+
+     * Documentation - Order is kept, when using block parsing it will be striped.
+     * Imports - Sorted and modified by isort - https://github.com/PyCQA/isort.
+     * General syntax lines - Order is kept, when using block parsing it will be striped.
+
+     The only way to control the structure is by using the following methods:
+
+     * add_line
+     * add_lines
+     * add_syntax_block
+     * add_variable
+     * extend
+     * insert
+     * syntax
+
+     Methods for adding documentation:
+
+     * add_doc_line
+     * add_doc_lines
+     * add_doc_block
+
+     Methods for adding imports:
+
+     * add_import
+     * add_imports
+     * add_specific_import
+
+     Methods for adding Python syntax:
+
+     * add_syntax_line
+     * add_syntax_lines
+     * add_syntax_block
+     * add_variable
+     * extend
+     * insert
+
+     Methods for getting the generated syntax:
+
+     * syntax
+     * use_black
+
+     Other Methods:
+
+     * validate
+     * parse_line
+
+    Args:
+        indentation (int): Determines how many spaces are used in the syntax indentation.
+    """
+
+    def __init__(
+        self,
+        indentation: int = 4,
+    ) -> None:
+        self._indentation = indentation * " "
+
+        self._docs: List[str] = []
+        self._imports: Set[str] = set()
+        self._lines: List[str] = []
+
+    def parse_block(self, block: str, new_line: int = 0, indent: int = 0) -> str:
+        """
+        Parsing a given block to be in the proper indentation and place.
+
+        Args:
+            block (str): A string that represents multiple line .
+            new_line (int): How many empty new lines to insert before the block.
+            indent (indent): How many indentations to insert before each line.
+
+        Returns:
+            str: The parsed line.
+        """
+        syntax = "\n" * new_line
+        syntax += re.sub("(^|\n)(.)", f"\\1{self._indentation * indent}\\2", block)
+        return syntax
+
+    def add_doc_line(self, line: str, indent: int = 0) -> "Block":
+        """
+        Adds a documentation line to the Python block syntax.
+
+        Args:
+            line (str): Documentation line of the block.
+            indent (int): How much to indent the syntax.
+
+        Returns:
+            Block: The block instance.
+        """
+        doc = self.parse_block(line, indent=indent)
+        self._docs.append(doc)
+        return self
+
+    def add_doc_lines(self, *lines: str, indent: int = 0) -> "Block":
+        """
+        Adds documentation lines to the Python block syntax.
+
+        Args:
+            *lines (Collection[str]): Documentation lines of the block.
+            indent (int): How much to indent the syntax.
+
+        Returns:
+            Block: The block instance.
+        """
+        add_doc_line = partial(self.add_doc_line, indent=indent)
+        tuple(map(add_doc_line, lines))
```

Comment:
You can use `list` instead of `tuple` (also faster a bit):
![image](https://user-images.githubusercontent.com/35102691/123049036-c29f2000-d407-11eb-9870-3215b922f14c.png)

Alternatively:
![image](https://user-images.githubusercontent.com/35102691/123049173-f2e6be80-d407-11eb-9e01-34e6c013b01a.png)